build.gradle

```diff
@@ -11,11 +11,55 @@ repositories {
 }
 
 dependencies {
-    implementation "org.jetbrains.kotlin:kotlin-stdlib"
-    testImplementation 'org.junit.jupiter:junit-jupiter-api:5.8.1'
-    testRuntimeOnly 'org.junit.jupiter:junit-jupiter-engine:5.8.1'
+    implementation("org.jetbrains.kotlin:kotlin-stdlib")
+    implementation("org.apache.maven:maven-artifact")
+    implementation("io.github.microutils:kotlin-logging-jvm:2.1.21")
+
+    testImplementation("org.junit.jupiter:junit-jupiter-api")
+    testImplementation("org.assertj:assertj-core")
+
+    testImplementation("org.junit.jupiter:junit-jupiter-params")
+
+    testRuntimeOnly("org.junit.jupiter:junit-jupiter-engine")
+
+    constraints {
+        api("org.apache.maven:maven-artifact:3.5.0")
+        api("org.junit.jupiter:junit-jupiter-api:$junitJupiterVersion")
+        api("org.junit.jupiter:junit-jupiter-engine:$junitJupiterVersion")
+        api("org.assertj:assertj-core:3.22.0")
+    }
+}
+
+tasks.named("compileTestKotlin").configure {
+    kotlinOptions {
+        freeCompilerArgs += "-opt-in=kotlin.io.path.ExperimentalPathApi"
+    }
 }
 
 test {
     useJUnitPlatform()
+    inputs.dir(layout.projectDirectory.dir("./src/test/fixtures"))
+}
+
+java {
+    sourceCompatibility = JavaVersion.VERSION_1_8
+    targetCompatibility = JavaVersion.VERSION_1_8
+    withSourcesJar()
+}
+
+gradlePlugin {
+    plugins {
```

Comment:
So I checked and since the configuration is not running in parallel and when a plugin is applied its `apply` method is executed right away, there should not be a problem with ensuring the correct order.

Even the official `JavaLibraryPlugin` relies on the applied `JavaPlugin` to create the SourceSet `main`: https://github.com/gradle/gradle/blob/master/subprojects/plugins/src/main/java/org/gradle/api/plugins/JavaLibraryPlugin.java